src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -560,8 +560,8 @@ String toExternalString() {
                 default:
                     throw new RuntimeException("Unhandled case in switch statement");
             }
-            String rest = compositeStrings.join(fields);
-            return (dynamoDbIterator == null ? streamArn : dynamoDbIterator) + ITERATOR_SEPARATOR + rest;
+            return (Objects.requireNonNullElse(dynamoDbIterator, streamArn))
+                + ITERATOR_SEPARATOR + compositeStrings.join(fields);
```

Comment:
Can be replaced with Objects.requireNonNullElse

Reply:
are the parentheses around `Objects.requireNonNullElse(dynamoDbIterator, streamArn)` necessary?

Reply:
fixed in 5f936f30727bd424a445bf64904d472d19170b76